core/api/src/main/java/com/microsoft/gctoolkit/jvm/JavaVirtualMachine.java

```diff
@@ -91,7 +93,8 @@
      * {@code Aggregation}s are used depends on the GC.
      * @param aggregationClass The class of the Aggregation to be returned.
      * @param <T> type cast for the Aggregation class type.
-     * @return an {@code Aggregation} whose {@code getClass() == aggregationClass}, or {@code null}
+     * @return an {@code Aggregation} whose {@code getClass() == aggregationClass}, or {@code null} 
+     * if given aggregationClass is not available.
      */
-    <T extends Aggregation> T getAggregation(Class<T> aggregationClass);
+    <T extends Aggregation> Optional<T> getAggregation(Class<T> aggregationClass);
```

Comment:
Maybe. I think Optional only really helps if you are using this API in a stream. Do you have a use-case/example for this?

Reply:
Whenever the return value may be null, I always prefer to see an Optional<T> so I know it may be null without having to even read the javadoc.

Reply:
It's either this or adding a method `hasAggregation(Class<T>)`.

Reply:
Fair enough.

Reply:
I like the enthusiasm but this thread is now mixing a couple of different concerns that is complicating the ability to discuss each of them. On point 1, one of the issues with Censum is that adding queries violated Open-Closed. Using the module version of the SPI solves this issue. To undo this features moves us back to having to violate Open-Closed. Again, SPI isn't magic, it's a well known pattern for providing implementations for an API.
On point 2, I'd like more discussion before we start renaming things. There is a model in play and I think that we should take more time to understand it before changing it.